src/border_agent/border_agent.cpp

```diff
@@ -205,6 +206,24 @@ static uint64_t ConvertTimestampToUint64(const otTimestamp &aTimestamp)
            static_cast<uint64_t>(aTimestamp.mAuthoritative);
 }
 
+void AppendOmrTxtEntry(otInstance &aInstance, Mdns::Publisher::TxtList &aTxtList)
```

Comment:
Similar to `AppendOmrTxtEntry`, I'm planning to wrap the rest of code of constructing TXT entries to helper functions.